File: python/spec/aoc2025/template.py
#!/usr/bin/env python3
#
from mamba import description, context, it
from expects import expect, equal

import spec.support
from aoc2025 import Classname

with description('Classname'):
  with context('Sample'):
    with before.all:
      self.subject = Classname(
          spec.support.file_to_list(__file__.replace('spec.py', 'sample.txt')))

    with description('Part 1'):

      with it('sample data agrees'):
        expect(self.subject.part_one()).to(equal(3))

    with description('Part 2'):

      with it('sample data agrees'):
        expect(self.subject.part_two()).to(equal(6))


  with context('Actual'):
    with before.all:
      self.subject = Classname(
          spec.support.file_to_list(__file__.replace('spec.py', 'actual.txt')))

    with description('Part 1'):

      with it('actual answer above'):
        print(self.subject.part_one())

      with it('results survive'):
        expect(self.subject.part_one()).to(equal(1139))

    with description('Part 2'):

      with it('actual answer above'):
        print(self.subject.part_two())

      with it('results survive'):
        expect(self.subject.part_two()).to(equal(6684))

File: ruby/spec/aoc2025/template.rb
module Aoc2025
  RSpec.describe Classname do
    context "Sample" do
      before(:all) do
        @subject = Classname.new Support.file_to_list __FILE__.gsub('spec.rb', 'sample.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "dunsels" do
        end

        it "sample data agrees" ,:skip do
          expect(subject.part_one).to eq 1
        end
      end

      describe "Part 2" do
        it "sample data agrees" ,:skip do
          expect(subject.part_deux).to eq 2
        end
      end
    end

    context "Actual" ,:skip do
      before(:all) do
        @subject = Classname.new Support.file_to_list __FILE__.gsub('spec.rb', 'actual.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "actual answer above" do
          pp subject.part_one
        end

        it "results survive" do
          expect(subject.part_one).to eq 1111
        end
      end

      describe "Part 2" ,:skip do
        it "actual answer above" do
          pp subject.part_deux
        end

        it "results survive" do
          expect(subject.part_deux).to eq 2222
        end
      end
    end
  end
end

